script/cmake-build

```diff
@@ -225,6 +225,19 @@ Could not fild the Android NDK CMake toolchain file
                 "-DOT_DNS_CLIENT_OVER_TCP=ON"
                 "-DOT_UDP_FORWARD=ON"
             )
+
+            local -r version_1_1_pattern_nonbsd='\-DOT_THREAD_VERSION=1\.1\>'
+            local -r version_1_1_pattern_bsd='-DOT_THREAD_VERSION=1\.1[[:>:]]'
+
+            if [[ $* =~ $version_1_1_pattern_nonbsd || $* =~ $version_1_1_pattern_bsd ]]; then
```

Comment:
This seems pretty hacky. Can we just always set `OT_CSL_RECEIVER=ON` and just not have it be a no-op on a Thread 1.1 build?

Reply:
@abtink introduced tristate option for OT. Setting `OT_CSL_RECEIVER=ON` is explicitly enable this feature, and if a user also explicitly set Thread version. It's invalid config, which should result in errors. This check may look hacky. Given $* contains all user inputs, it may not result in unexpected behavior.

Another option is we define 1.2+ feature options with [CMakeDependentOption](https://cmake.org/cmake/help/v3.10/module/CMakeDependentOption.html#module:CMakeDependentOption).

Reply:
This change doesn't make these options not configurable. Actually it's just providing default value. CMake allows the same option be provided multiple times, and the last instance succeeds. With this change, OT_CSL_RECEIVER is ON by default in 1.2+, and we can still explicitly disable it by appending as @abtink suggested approach.